lib/config/global.ts

```diff
@@ -40,7 +40,10 @@ export class GlobalConfig {
 
     const result = { ...config };
     for (const option of GlobalConfig.OPTIONS) {
-      GlobalConfig.config[option] = config[option];
+      GlobalConfig.config = {
```

Comment:
Do we really need to clone that object here?

Reply:
I didn't manage to make types work another way, i.e. by assignment

Reply:
When I switch to assignment, it's not able to track the field type properly

Reply:
I think we can rely on some unsafe behavior here, as we know the types will match.

Recreating the object in the loop is pretty bad if it's only for type check.

Reply:
Agree, trying to please type system, I went too far